tango/step.py

```diff
@@ -115,6 +115,7 @@ def __init__(
         cache_results: Optional[bool] = None,
         step_format: Optional[Format] = None,
         step_config: Optional[Dict[str, Any]] = None,
+        step_unique_id: Optional[str] = None,
```

Comment:
This needs documentation. And maybe let's call it `step_unique_id_override` to make it a little clearer what the relationship is between this value and the step's unique ID.